client.go

```diff
@@ -326,19 +342,19 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 
 	response := &Response{Response: resp}
 
-	err = CheckResponse(resp)
+	ok, err := CheckResponse(resp, errorResponse)
 	if err != nil {
 		return response, err
 	}
 
-	if v != nil {
-		if w, ok := v.(io.Writer); ok {
+	if ok && successResponse != nil {
```

Comment:
```suggestion
	if successResponse != nil {
```
aqui a gente pode remover o `ok`, já que se der erro no `CheckResponse` vai ser retornado logo abaixo dele na verificação.